testng-core/src/test/java/test/configuration/ConfigurationGroupsTest.java

```diff
@@ -37,4 +39,26 @@ public void runTest(Class<?> testClass, String groupName) {
       {ConfigurationGroups7SampleTest.class, "A"}
     };
   }
+
+  @Test(description = "GITHUB-2664", dataProvider = "getTestClasses")
+  public void ensureDependsOnGroupsHonouredForConfigs(Class<?> clazz) {
+    MethodInvocationListener.logs.clear();
+    TestNG testng = create(clazz);
+    testng.setVerbose(2);
+    testng.setListenerClasses(Collections.singletonList(MethodInvocationListener.class));
```

Comment:
If you addListener instance instead of the listener class, you'll be able to get log from instance instead of dealing with static.

Reply:
Fixed.